apis/tempo/v1alpha1/tempostack_types.go

```diff
@@ -387,6 +389,48 @@ type ObjectStorageTLSSpec struct {
 	CA string `json:"caName,omitempty"`
 }
 
+// ReceiversTLSSpec is the TLS configuration for the receivers.
+type ReceiversTLSSpec struct {
+	Enabled bool `json:"enabled"`
+
+	// caName is the name of a ConfigMap containing a CA certificate.
+	// It needs to be in the same namespace as the Tempo custom resource.
+	//
+	// +required
+	// +kubebuilder:validation:Required
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:io.kubernetes:ConfigMap",displayName="CA ConfigMap Name"
+	CA    string `json:"caName,omitempty"`
+	CAKey string `json:"caKey,omitempty"`
+	// certName is the name of a Secret containing a certificate and the private key
+	// It needs to be in the same namespace as the Tempo custom resource.
+	//
+	// +required
+	// +kubebuilder:validation:Required
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:io.kubernetes:Secret",displayName="Certificate Secret Name"
+	Cert string `json:"certName,omitempty"`
+	// certName is the name of a Secret containing a certificate and the private key
+	// It needs to be in the same namespace as the Tempo custom resource.
+	//
+	// +optional
+	// +kubebuilder:validation:optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Key of the secret for private key"
+	CertPrivateKey string `json:"certPrivateKey,omitempty"`
```

Comment:
Personally I'd hardcode `tls.key` and `tls.crt`, afaics it's a Kubernetes convention to use these key names: https://kubernetes.io/docs/concepts/configuration/secret/#tls-secrets
But I don't have a very strong opinion on that.

Reply:
(and also hardcode `ca.crt`, but I couldn't find a convention for that)

Reply:
Make sense to me, I haven't found also something for the ca, but I see other parts of this code use ca.crt so I think we will be fine.